deduplicator/src/lib/db_hashes.rs

```diff
@@ -0,0 +1,551 @@
+//! Definitions of the structure of a working database to save addresses and how to interact with
+//! it. This database will be used to save hashes of all imported addresses and compute collisions
+//! between them.
+
+use std::convert::TryInto;
+use std::path::PathBuf;
+
+use rusqlite::{Connection, Statement, ToSql, Transaction, NO_PARAMS};
+use tools::Address;
+
+/// Name of the table containing addresses.
+const TABLE_ADDRESSES: &str = "addresses";
+
+/// Name of the table containing hashes. For each address, multiple hashes may be stored.
+const TABLE_HASHES: &str = "_addresses_hashes";
+
+/// Name of the table listing addresses that have to be removed to eliminate all duplicates.
+const TABLE_TO_DELETE: &str = "_to_delete";
+
+//  ____  _     _   _           _
+// |  _ \| |__ | | | | __ _ ___| |__   ___  ___
+// | | | | '_ \| |_| |/ _` / __| '_ \ / _ \/ __|
+// | |_| | |_) |  _  | (_| \__ \ | | |  __/\__ \
+// |____/|_.__/|_| |_|\__,_|___/_| |_|\___||___/
+//
+
+/// A database, this structure can be used to open connections or perform high-level operations.
+pub struct DbHashes {
+    db_path: PathBuf,
+}
+
+impl DbHashes {
+    /// Instantiate a new database from a path to an SQLite file.
+    ///
+    /// If the file is not created yet or if the schema is not already set up, this will be done.
+    ///
+    /// # Examples
+    /// ```
+    /// # use std::fs::remove_file;
```

Comment:
Instead of all this, just use "no_run" on your code example. It makes unnecessary computations.